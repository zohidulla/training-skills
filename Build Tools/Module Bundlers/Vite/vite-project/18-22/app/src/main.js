import { pluckAndLog } from "@zohidjon/pluck";
import { log } from "@zohidjon/pluck/log";

const users = [
  {
    name: "Ali",
    age: 34,
  },
  {
    name: "Vali",
    age: 25,
  },
  {
    name: "Bek",
    age: 26,
  },
];

pluckAndLog(users, "name");
