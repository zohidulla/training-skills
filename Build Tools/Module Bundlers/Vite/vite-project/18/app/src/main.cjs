const { pluck } = require("pluck");
const { log } = require("pluck/log");

const users = [
  {
    name: "Ali",
    age: 34,
  },
  {
    name: "Vali",
    age: 25,
  },
  {
    name: "Bek",
    age: 26,
  },
];

log(pluck(users, "name"));
