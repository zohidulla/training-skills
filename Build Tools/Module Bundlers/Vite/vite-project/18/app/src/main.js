import { pluck } from "pluck";

const users = [
  {
    name: "Ali",
    age: 34,
  },
  {
    name: "Vali",
    age: 25,
  },
  {
    name: "Bek",
    age: 26,
  },
];

console.log(pluck(users, "name"));
