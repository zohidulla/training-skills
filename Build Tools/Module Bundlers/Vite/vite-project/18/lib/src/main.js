export function pluck(collection, field) {
  return collection.map((item) => item[field]);
}

// const users = [
//   {
//     name: "John",
//     age: 34,
//   },
//   {
//     name: "Jane",
//     age: 25,
//   },
//   {
//     name: "Alex",
//     age: 26,
//   },
// ];

// console.log(pluck(users, "name"));
