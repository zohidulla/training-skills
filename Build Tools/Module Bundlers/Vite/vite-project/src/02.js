import collect from "collect.js";

console.log(collect([1, 2, 3, 4, 5]).avg());
