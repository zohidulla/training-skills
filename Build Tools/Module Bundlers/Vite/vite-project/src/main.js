import data from "./assets/07.json";

console.log(data);
