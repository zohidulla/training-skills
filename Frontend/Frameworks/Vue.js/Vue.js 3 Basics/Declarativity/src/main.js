import "./style.css";
import { watchEffect, ref } from "vue";
import { shuffle } from "lodash";
import morphdom from "morphdom";

const SIZE = 60;
const colors = ref([
  { r: 255, g: 0, b: 0 },
  { r: 0, g: 255, b: 0 },
  { r: 0, g: 0, b: 255 },
]);

watchEffect(() => {
  const existingContainer = document.querySelector("#colors");

  const container = document.createElement("div");
  container.id = "colors";

  colors.value
    .map((c, i) => {
      const color = `rgb(${c.r}, ${c.g}, ${c.b})`;
      const node = document.createElement("div");
      node.classList.add("color");
      node.id = color;
      node.style.backgroundColor = color;
      node.style.borderColor = `rgb(${c.r * 0.8}, ${c.g * 0.8}, ${c.b * 0.8})`;
      node.style.transform = `translateX(${SIZE * i}px)`;
      return node;
    })
    .sort((node1, node2) => (node1.id > node2.id ? 1 : -1))
    .forEach((node) => {
      container.appendChild(node);
    });

  morphdom(existingContainer, container);
});

document.querySelector("button#add").addEventListener("click", () => {
  colors.value.push({
    r: Math.random() * 255,
    g: Math.random() * 255,
    b: Math.random() * 255,
  });
});

document.querySelector("button#reset").addEventListener("click", () => {
  colors.value = [];
});

document.querySelector("button#shuffle").addEventListener("click", () => {
  colors.value = shuffle(colors.value);
});
