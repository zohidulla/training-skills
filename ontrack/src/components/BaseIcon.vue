<script setup>
import {
  CheckCircleIcon as CheckCircle,
  ListBulletIcon as ListBullet,
  ArrowPathIcon as ArrowPath,
  ChartBarIcon as ChartBar,
  ClockIcon as Clock,
  TrashIcon as Trash,
  XMarkIcon as XMark,
  PauseIcon as Pause,
  PlayIcon as Play,
  PlusIcon as Plus
} from '@heroicons/vue/24/outline'

defineProps({
  name: {
    required: true,
    type: String
  }
})

const icons = {
  CheckCircle,
  ListBullet,
  ArrowPath,
  ChartBar,
  Clock,
  Trash,
  XMark,
  Pause,
  Play,
  Plus
}
</script>

<template>
  <component :is="icons[name]" />
</template>
