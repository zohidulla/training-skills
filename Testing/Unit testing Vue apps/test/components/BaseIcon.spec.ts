import { expect, it } from 'vitest'
import { mount, shallowMount } from '@vue/test-utils'
import { IconName } from '../../src/types'
import BaseIcon from '../../src/components/BaseIcon.vue'
import { ICONS } from '../../src/icons'

it('renders icon with default classes', () => {
  const wrapper = shallowMount(BaseIcon, {
    props: { name: IconName.CLOCK }
  })

  expect(wrapper.classes()).toContain('h-8')
})

it('renders icon with custom classes', () => {
  const classes = 'h-12 text-purple-600'
  const wrapper = shallowMount(BaseIcon, {
    attrs: { class: classes },
    props: { name: IconName.CLOCK, size: 16 }
  })

  expect(wrapper.classes().join(' ')).toBe(classes)
})

it.each(Object.keys(ICONS))('renders %s icon', (name) => {
  const wrapper = mount(BaseIcon, {
    props: { name: name as IconName }
  })

  expect(wrapper.html()).toMatchSnapshot()
})
