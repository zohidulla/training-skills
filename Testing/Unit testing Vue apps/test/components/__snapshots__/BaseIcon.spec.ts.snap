// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders ArrowPath icon 1`] = `
"<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" aria-hidden="true" data-slot="icon" class="h-8">
  <path stroke-linecap="round" stroke-linejoin="round" d="M16.023 9.348h4.992v-.001M2.985 19.644v-4.992m0 0h4.992m-4.993 0 3.181 3.183a8.25 8.25 0 0 0 13.803-3.7M4.031 9.865a8.25 8.25 0 0 1 13.803-3.7l3.181 3.182m0-4.991v4.99"></path>
</svg>"
`;

exports[`renders ChartBar icon 1`] = `
"<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" aria-hidden="true" data-slot="icon" class="h-8">
  <path stroke-linecap="round" stroke-linejoin="round" d="M3 13.125C3 12.504 3.504 12 4.125 12h2.25c.621 0 1.125.504 1.125 1.125v6.75C7.5 20.496 6.996 21 6.375 21h-2.25A1.125 1.125 0 0 1 3 19.875v-6.75ZM9.75 8.625c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125v11.25c0 .621-.504 1.125-1.125 1.125h-2.25a1.125 1.125 0 0 1-1.125-1.125V8.625ZM16.5 4.125c0-.621.504-1.125 1.125-1.125h2.25C20.496 3 21 3.504 21 4.125v15.75c0 .621-.504 1.125-1.125 1.125h-2.25a1.125 1.125 0 0 1-1.125-1.125V4.125Z"></path>
</svg>"
`;

exports[`renders CheckCircle icon 1`] = `
"<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" aria-hidden="true" data-slot="icon" class="h-8">
  <path stroke-linecap="round" stroke-linejoin="round" d="M9 12.75 11.25 15 15 9.75M21 12a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"></path>
</svg>"
`;

exports[`renders Clock icon 1`] = `
"<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" aria-hidden="true" data-slot="icon" class="h-8">
  <path stroke-linecap="round" stroke-linejoin="round" d="M12 6v6h4.5m4.5 0a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"></path>
</svg>"
`;

exports[`renders ListBullet icon 1`] = `
"<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" aria-hidden="true" data-slot="icon" class="h-8">
  <path stroke-linecap="round" stroke-linejoin="round" d="M8.25 6.75h12M8.25 12h12m-12 5.25h12M3.75 6.75h.007v.008H3.75V6.75Zm.375 0a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0ZM3.75 12h.007v.008H3.75V12Zm.375 0a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0Zm-.375 5.25h.007v.008H3.75v-.008Zm.375 0a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0Z"></path>
</svg>"
`;

exports[`renders Pause icon 1`] = `
"<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" aria-hidden="true" data-slot="icon" class="h-8">
  <path stroke-linecap="round" stroke-linejoin="round" d="M15.75 5.25v13.5m-7.5-13.5v13.5"></path>
</svg>"
`;

exports[`renders Play icon 1`] = `
"<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" aria-hidden="true" data-slot="icon" class="h-8">
  <path stroke-linecap="round" stroke-linejoin="round" d="M5.25 5.653c0-.856.917-1.398 1.667-.986l11.54 6.347a1.125 1.125 0 0 1 0 1.972l-11.54 6.347a1.125 1.125 0 0 1-1.667-.986V5.653Z"></path>
</svg>"
`;

exports[`renders Plus icon 1`] = `
"<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" aria-hidden="true" data-slot="icon" class="h-8">
  <path stroke-linecap="round" stroke-linejoin="round" d="M12 4.5v15m7.5-7.5h-15"></path>
</svg>"
`;

exports[`renders Trash icon 1`] = `
"<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" aria-hidden="true" data-slot="icon" class="h-8">
  <path stroke-linecap="round" stroke-linejoin="round" d="m14.74 9-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 0 1-2.244 2.077H8.084a2.25 2.25 0 0 1-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 0 0-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 0 1 3.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 0 0-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 0 0-7.5 0"></path>
</svg>"
`;

exports[`renders XMark icon 1`] = `
"<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" aria-hidden="true" data-slot="icon" class="h-8">
  <path stroke-linecap="round" stroke-linejoin="round" d="M6 18 18 6M6 6l12 12"></path>
</svg>"
`;
